eth2near/contract_wrapper/src/eth_network_enum.rs

```diff
@@ -0,0 +1,63 @@
+use std::error::Error;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+use serde::Deserialize;
+
+#[derive(Debug, Clone, Deserialize)]
+pub enum EthNetwork {
+    Mainnet,
+    Kiln,
+    Ropsten,
+    Goerli,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct IncorrectEthNetwork;
+
+impl Display for IncorrectEthNetwork {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Unknown Eth network. Possible Eth network: 'Mainnet', 'Kiln', 'Goerli', 'Ropsten'"
+        )
+    }
+}
+
+impl Error for IncorrectEthNetwork {}
+
+impl Display for EthNetwork {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            EthNetwork::Mainnet => write!(f, "mainnet"),
+            EthNetwork::Kiln => write!(f, "kiln"),
+            EthNetwork::Goerli => write!(f, "goerli"),
+            EthNetwork::Ropsten => write!(f, "ropsten")
+        }
```

Comment:
No need to duplicate the names of networks as `as_str()` is already implemented.

```suggestion
        write!(f, "{}", self.as_str())
```

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/e4e2d1194f7d4e6811452999d30c08b4ec39e34e